crates/adapters/src/test/iceberg.rs

```diff
@@ -0,0 +1,372 @@
+inputs:
+    test_input1:
+        stream: test_input1
+        transport:
+            name: "iceberg_input"
+            config:
+{}
+"#,
+        options,
+    );
+
+    let config: PipelineConfig = serde_yaml::from_str(&config_str).unwrap();
+    let schema = schema.to_vec();
+
+    let (err_sender, err_receiver) = crossbeam::channel::unbounded();
+
+    let controller = Controller::with_config(
+        move |workers| Ok(test_circuit::<T>(workers, &schema)),
+        &config,
+        Box::new(move |e| {
+            let msg = format!("iceberg_input_test: error: {e}");
+            println!("{}", msg);
+            err_sender.send(msg).unwrap()
+        }),
+    )
+    .unwrap();
+
+    (controller, err_receiver)
+}
+
+/// Generate up to `max_records` _unique_ records.
+#[cfg(feature = "iceberg-tests-fs")]
+fn data(n_records: usize) -> Vec<TestStruct2> {
+    let mut result = Vec::with_capacity(n_records);
+
+    let mut time =
+        chrono::NaiveDateTime::parse_from_str("2024-01-01 00:00:00", "%Y-%m-%d %H:%M:%S").unwrap();
+
+    for i in 0..n_records {
+        result.push(TestStruct2 {
+            field: i as i64,
+            field_0: Some(format!("Name_{i}")),
+            field_1: true,
+            field_2: feldera_sqllib::Timestamp::from_naiveDateTime(time),
+            field_3: feldera_sqllib::Date::from_date(time.date()),
+            field_5: None,
+            field_6: None,
+        });
+
+        time += std::time::Duration::from_secs(1);
+    }
+
+    result
+}
+
+#[test]
+#[cfg(feature = "iceberg-tests-fs")]
+fn iceberg_localfs_input_test_unordered() {
+    iceberg_localfs_input_test(&[], &|_| true);
+}
+
+#[test]
+#[cfg(feature = "iceberg-tests-fs")]
+fn iceberg_localfs_input_test_ordered() {
+    iceberg_localfs_input_test(
+        &[("timestamp_column".to_string(), "ts".to_string())],
+        &|_| true,
+    );
+}
+
+#[test]
+#[cfg(feature = "iceberg-tests-fs")]
+fn iceberg_localfs_input_test_ordered_with_filter() {
+    iceberg_localfs_input_test(
+        &[
+            ("timestamp_column".to_string(), "ts".to_string()),
+            ("snapshot_filter".to_string(), "id >= 10000".to_string()),
+        ],
+        &|x| x.field >= 10000,
+    );
+}
+
+#[cfg(feature = "iceberg-tests-fs")]
+fn iceberg_localfs_input_test(
+    extra_config: &[(String, String)],
+    filter: &dyn Fn(&TestStruct2) -> bool,
+) {
+    let data = data(1_000_000);
+
+    let table_dir = tempfile::TempDir::new().unwrap();
+    let table_path = table_dir.path().display().to_string();
+
+    let ndjson_file = data_to_ndjson(data.clone());
+    println!("wrote test data to {}", ndjson_file.path().display());
+
+    // // Uncomment to inspect output parquet files produced by the test.
+    // std::mem::forget(table_dir);
+
+    let script_path = "../iceberg/src/test/create_test_table_s3.py";
+
+    // Run the Python script using the Python interpreter
+    let output = std::process::Command::new("python3")
```

Comment:
this can't be done in rust? I already lost track on how to run adapter tests

Reply:
No way to write Iceberg in Rust yet, that's why the script.

>  I already lost track on how to run adapter tests

that's why all iceberg tests are feature gated and don't run by default